components/brave_wallet/browser/brave_wallet_provider.cc

```diff
@@ -0,0 +1,42 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/brave_wallet_provider.h"
+
+#include <utility>
+
+#include "brave/components/brave_wallet/brave_wallet_service.h"
+
+namespace brave_wallet {
+
+BraveWalletProvider::BraveWalletProvider(BraveWalletService* wallet_service)
+    : wallet_service_(wallet_service), weak_factory_(this) {}
+
+BraveWalletProvider::~BraveWalletProvider() {}
+
+void BraveWalletProvider::Request(const std::string& json_payload,
+                                  RequestCallback callback) {
+  if (!wallet_service_)
```

Comment:
this should never be null here because we shouldn't be creating `BraveWalletProvider` in the first place if it is

Reply:
the check should stay as I remade it to have a WeakPtr in case of IPC message comes but service is already dead